### PR TITLE
Maint/isort

### PR DESCRIPTION
**Pull request by mschmidt87**
This PR introduces [isort](https://github.com/timothycrosley/isort) to our tools ensuring common coding standards and uniform layout across the library.

`isort` sorts import statements in a particular, reproducible order, namely like this: `FUTURE,STDLIB,THIRDPARTY,FIRSTPARTY,LOCALFOLDER`

This PR:
- applies `isort` to all files in the library, testsuite, and examples. Therefore there are a lot of changes in many files, but they are just about formatting, mostly to the imports. Some changes are introduced by `black` as well (trailing commas in function definitions). I am not sure why this happens.
- adds a config file for `isort`
- adds `isort` to the required `dev` libraries
- adds `isort` as an additional check to the CI.
- adds my pre-commit configuration file to the repository. This will make it easier for other contributors to install the pre-commit hooks using that config file.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+repos:
+-   repo: https://github.com/ambv/black
+    rev: 19.3b0
```
Comment (jakobj) [inline .pre-commit-config.yaml:3]: how will we go about maintaining these versions? would it be possible to put a "most recent" here?

File: .travis.yml [hunk 2]
```diff
@@ -30,6 +30,7 @@ script:
   - black --check .
   - flake8 --config=.flake8 .
   - mypy cgp
+  - isort --check-only cgp
```
Comment (jakobj) [inline .travis.yml:33]: should we check examples and tests as well? why restrict this to the library path?

File: .pre-commit-config.yaml (hunk 1 above)
Comment (mschmidt87) [reply .pre-commit-config.yaml:3]: we can put a stable for black but there is no stable tag for the other tools. Honestly, I think it is not that relevant to care about the versions here.

File: .travis.yml (hunk 2 above)
Comment (mschmidt87) [reply .travis.yml:33]: yes, totally, thanks for noticing, I added that.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (jakobj) [reply .pre-commit-config.yaml]: ok

**Comment by jakobj** [comment]
before we merge, please squash the commits

**Comment by mschmidt87** [comment]
Thanks, I organized this now into 5 commits, which makes sense, in my opinion. Objections, @jakobj ?

**Comment by jakobj** [comment]
i'm happy! merging :)

**Pull request merged**